modules/reactor/src/meshgenerators/ReactorMeshParams.C

```diff
@@ -43,8 +43,12 @@ ReactorMeshParams::validParams()
   params.addParam<std::vector<unsigned int>>(
       "axial_mesh_intervals",
       "Number of elements in the Z direction for each axial region");
+  params.addPrivateParam<bool>("_bypass_meshgen", true);
```

Comment:
The logic for this is a bit convoluted. The way I've code the optimization to be triggered is 

1. Is `Mesh/data_driven_generator` set by user? If so, check 2 as well
2. Has the value of private parameter `_bypass_meshgen` set to true?

The second conditional is added purely for the the case when an external app like Griffin calls `ReactorMeshParam` as a sub-generator but still wants an output mesh (the homogenized mesh) to be generated, even though `Mesh/data_driven_generator` has been set for the heterogeneous mesh.

Reply:
Are you thinking about the case where `ReactorMeshParams` might be called in succesion in the same input? I wasn't thinking about that specific case but I guess we can add that check. Do you know how I can check if data driven generator has been called upstream?

Reply:
i.e. getting the tree of generators that has been called so far?

Reply:
> Do you know how I can check if data driven generator has been called upstream?

no I dont. Hopefully there s an API in the MeshGeneratorSystem header

Reply:
no I dont see one.
Should be something we can do there.
If you dont want to do it, I d still like the default to be the safe one. 
and some sort of paragraph in comments explaining what we want to do there and why

Reply:
Alternatively, can we just remove the default value from `_bypass_meshgen` and only read it when it has a valid value? This is a private parameter so users can't modify it directly, the only way it can be set is through a sub generator call. I've updated `ReactorMeshParams` with this new logic and added a description of when sub generator calls would want to manually set this value

Reply:
The default behavior will still be dictated by whether or not `data_driven_generator` exists in the `[Mesh]`  block

Reply:
It s just weird to have a default of true when in reality this parameter is ignored.

I think if we want to do this via a parameter maybe it should be:
- default to false
- when setting a data_driven_generator, for all the generators upstream of it, set to true by default

I know I m not super clear on this, as I keep switching between tasks here

Reply:
Yeah the private parameter was added as a workaround for the case when `ReactorMeshParams` is called as a sub generator or not. I couldn't find a way to determine whether a specific ReactorMeshParams is a sub generator call (should generate mesh) or not (skip mesh generation). I'm hesitant to make changes to framework level for this very specific case, in case it ends up being misused for some other application

In any case, is the dependency tree of which generators are upstream / downstream known at the constructor level, or is this determined during `generate`? I'm basically trying to set this logic at construction time

Reply:
>  is the dependency tree of which generators are upstream / downstream known at the constructor level

I dont know. should be easy to check. Add a print statement in two constructors and re-order them in the input file.

Reply:
Looks like switching the order of two mesh generators where one is dependent on the other doesn't still calls the MG's in the proper order so the dependencies are probably known at construction time. I'll have to investigate the framework code a bit more closely to see if there's a way to figure out how to bypass mesh generation without using an additional private parameter

Reply:
You can add the private parameter if it's simpler that way. It just is better to set it dynamically to the right value than a blanket `skip`

Reply:
@GiudGiud the variable `ordered_generators` in `MeshGeneratorSystem::createAddedMeshGenerators` stores the ordering of MG's in the input file. Do you think it makes sense to have a function `isUpstreamGenerator(current_generator, generator_to_compare)` which returns True if `current_generator` comes after `generator_to_compare` in `ordered_generators`, and False otherwise? Keep in mind `ordered_generators` does not store any sub generator names (and it doesn't seem like there's a way to know what the sub generators will be in the constructor stage), so this would return False in this case. This function then can be used to define `_bypass_meshgen` private parameter.

Reply:
@GiudGiud I've implemented a function `MeshGenerator::isUpstreamGenerator` which checks whether the input argument is defined upstream of the current mesh generator (see latest commit), which requires framework changes. This allows me to remove the private variable entirely and dynamically determine whether the `ReactorMeshParams` generator is defined upstream of `Mesh/data_driven_generator`.  If ReactorMeshParams generator occurs before data_driven_generator, we bypass mesh generation. If ReactorMeshParams generator occurs after the data_driven_generator or is a sub generator, we generate a mesh.